libbeat/autodiscover/providers/kubernetes/node.go

```diff
@@ -168,6 +168,11 @@ func (n *node) emit(node *kubernetes.Node, flag string) {
 		return
 	}
 
+	// If the node is not in ready state then dont monitor it
+	if !isNodeReady(node) {
```

Comment:
What happen when the nodes is removed? Should n't that check be like:
```suggestion
	if !isNodeReady(node) && flag != "stop" {
```
so as to remove the node's config anyways?

Reply:
done

Reply:
Thanks!